### ISSUE TITLE
Matrix resizeKeep() should be smarter about space allocation

### ISSUE THREAD

**Issue by sherm1** [simbody/simbody#401]
This gets used sometimes to build up a Matrix row by row or column by column. In that case it should act like std::vector and preallocate extra space when it runs out. Currently it just allocates what is asked for then copies the old data so naive use has O(n^2) complexity for adding n rows or columns.

@klshrinidhi ran into this for the DataTable implementation in OpenSim.

**Comment by ianhamilton117** [comment]
@sherm1 Hi there, I'm new to Simbody and am looking for an issue to start contributing. Can I take this one?

**Comment by chrisdembia** [comment]
Hey @ianhamilton117. No one is working on this issue currently, so if you think you know how to do this, we would welcome the contribution.

**Comment by sherm1** [comment]
Agreed, thanks @ianhamilton117! Please be sure to include a unit test that can run in our automated builds, verifies that the code is working, and ensures that it continues to work after some future programmer mangles it.

**Comment by chrisdembia** [comment]
@ianhamilton117 to ensure a swift review of your changes, you may want to look over [CONTRIBUTING.md](https://github.com/simbody/simbody/blob/master/CONTRIBUTING.md).

**Comment by ianhamilton117** [comment]
Thanks @chrisdembia, will do.
